eth_utils/address.py

```diff
@@ -145,38 +135,42 @@ def is_normalized_address(value):
         return value == to_normalized_address(value)
 
 
-@coerce_args_to_bytes
-@coerce_return_to_bytes
-def to_canonical_address(address):
+def to_canonical_address(primitive=None, hexstr=None, text=None):
     """
+    Given any supported representation of an address
+    returns it's canonical form (20 byte long string).
     """
-    return decode_hex(to_normalized_address(address))
+    text_address = to_text(primitive, hexstr, text)
+    return decode_hex(to_normalized_address(text_address))
 
 
-def is_canonical_address(value):
-    if not is_address(value):
+def is_canonical_address(address):
```

Comment:
I'm not sure if this behaves well with the new `is_address()` ? I can't seem to get the test case from the readme to pass (address=`b'\xd3\xcd\xa9\x13\xde\xb6\xf6yg\xb9\x9dg\xac\xdf\xa1q,)6\x01xd'`)

Reply:
This is due to the change to the signature of `is_checksum_address` and `to_checksum_address`.  I think if you revert the changes to those function signatures you'll be on the right track to fixing them.

Reply:
I guess i'm a little confused on what exactly a canonical representation is. Isn't it a 20byte long string of bytes? @carver suggested in the last pr the below implementation and "We don't need to test the canonical-ness of bytes, it's clearly not. "
```python
def is_canonical_address(address):
    if not is_string(address) or not is_hex(address):
        return False
    elif not is_address(hexstr=address):
        return False
    else:
        return address == to_canonical_address(hexstr=address)
```
The test case from the readme passes now but breaks if I add `if not is_string(address) or not is_hex(address):`

Reply:
Canonical address definition is a byte-string of length 20.  Feel free to modify this function to be *loosely* the following.

```python
def is_canonical_address(value):
    return is_bytes(value) and len(value) == 20
```

Reply:
Ah, interesting, yes I misinterpreted `canonical`. The "typical" representation is checksum, so I was basically reading it as checking if the address is the checksum. I think the same things I was saying before still apply, you can just reject it if it's not `bytes`, rather than convert `str` to `bytes` and check if that `is_address`, etc.

Reply:
I'm thinking something like . . . ?
```python
if not is_bytes(address) or len(address) != 20:
    return False
return address == to_canonical_address(address)